src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,69 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+
 public class SalaryInfo {
+    private static final int ARRAY_INDEX_FOR_NAME = 1;
+    private static final int ARRAY_INDEX_FOR_WORKING_HOURS = 2;
+    private static final int ARRAY_INDEX_FOR_INCOME_PER_HOUR = 3;
+    private static final int BEGINNING_INDEX_FOR_DAY = 0;
+    private static final int ENDING_INDEX_FOR_DAY = 2;
+    private static final int BEGINNING_INDEX_FOR_MONTH = 3;
+    private static final int ENDING_INDEX_FOR_MONTH = 5;
+    private static final int BEGINNING_INDEX_FOR_YEAR = 6;
+    private static final int ENDING_INDEX_FOR_YEAR = 10;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int salary = 0;
+        StringBuilder salaryInfo = new StringBuilder();
+        try {
+            for (String name : names) {
+                if (name != null) {
+                    for (String datum : data) {
+                        String[] dataEntry = datum.split(" ");
+                        if (name.equals(dataEntry[ARRAY_INDEX_FOR_NAME])
+                                && isDatePass(datum, dateFrom, dateTo)) {
+                            salary += Integer.parseInt(dataEntry[ARRAY_INDEX_FOR_WORKING_HOURS])
+                                    * Integer.parseInt(dataEntry[ARRAY_INDEX_FOR_INCOME_PER_HOUR]);
+                        }
+                    }
+                    StringBuilder stringBuilder = new StringBuilder(name);
+                    if (name.equals(names[names.length - 1])) {
+                        salaryInfo.append(stringBuilder.append(" - ").append(salary));
+                    } else {
+                        salaryInfo.append(stringBuilder.append(" - ").append(salary))
+                                .append(System.lineSeparator());
+                    }
+                    salary = 0;
+                }
+            }
+        } catch (RuntimeException e) {
```

Comment:
we shouldn't catch RuntimeExceptions and it's subclasses, please remove try/catch block